OpenRA.Game/GameRules/MusicInfo.cs

```diff
@@ -39,12 +41,18 @@ public void Load()
 				return;
 
 			Exists = true;
-			using (var s = Game.ModData.ModFiles.Open(Filename))
+
+			using (var stream = Game.ModData.ModFiles.Open(Filename))
 			{
-				if (Filename.ToLowerInvariant().EndsWith("wav"))
-					Length = (int)WavLoader.WaveLength(s);
-				else
-					Length = (int)AudLoader.SoundLength(s);
+				foreach (var loader in Game.ModData.SoundLoaders)
```

Comment:
It would be much cleaner to identify the file type first, and then call GetLength for that type.